lib/modules/ai_bot/site_settings_extension.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module DiscourseAi::AiBot::SiteSettingsExtension
+  module ClassMethods
+    def ai_bot_enabled=(val)
+      super(val)
+      DiscourseAi::AiBot::SiteSettingsExtension.enable_or_disable_ai_bots
+    end
+
+    def ai_bot_enabled_chat_bots=(val)
+      super(val)
+      DiscourseAi::AiBot::SiteSettingsExtension.enable_or_disable_ai_bots
+    end
+  end
+
+  def self.enable_or_disable_ai_bots
+    enabled_bots = SiteSetting.ai_bot_enabled_chat_bots.split("|")
+    enabled_bots = [] if !SiteSetting.ai_bot_enabled
+    DiscourseAi::AiBot::EntryPoint::BOTS.each do |id, bot_name, name|
+      active = enabled_bots.include?(name)
+      user = User.find_by(id: id)
+
+      if active
+        if !user
+          user =
+            User.new(
+              id: id,
+              email: "no_email_#{name}",
+              name: bot_name.titleize,
+              username: UserNameSuggester.suggest(bot_name),
+              active: true,
+              approved: true,
+              admin: true,
+              moderator: true,
+              trust_level: TrustLevel[4],
+            )
+          user.save!(validate: false)
+        else
+          user.update!(active: true)
```

Comment:
I guess the only potential issue is that these bang! methods may throw an error, does this stuff happen in a job, or would it prevent the app from booting if e.g. this errored when the seed is run?

Reply:
should never run on seed (it calls outside of it) but yeah it has potential to be an issue, though downside of eating this is that we will not pick up on the error

Reply:
Yeah it's kind of tricky either way. If it won't prevent boot it's probably fine, though to be super duper safe it could be outsourced to a sidekiq job.